test/sync_tests/online.test.js

```diff
@@ -1012,4 +1012,29 @@ describe('online syncing', function () {
     expect(this.application.apiService.session.accessToken).to.be.ok;
     expect(this.application.apiService.session.refreshToken).to.be.ok;
   });
+
+  it('should not allow receiving decrypted payloads from server', async function () {
+    const masterCollection = this.application.payloadManager.getMasterCollection();
+    const historyMap = this.application.historyManager.getHistoryMapCopy();
+    const note = await Factory.createSyncedNote(this.application);
+    this.expectedItemCount++;
```

Comment:
this looks dangerous
why do this?

Reply:
If the sync server gets compromised and an attacker introduces a malicious payload to our items collection, this change would reject that payload as it couldn't have been encrypted with our local-only root key

Reply:
Ah, you're answering about the entire pull request. This I understand.
What I'm asking about here is the line with `this.expectedItemCount++`. Seems like it's changing some state which doesn't seem to have anything to do with the rest of the test.
What is `expectedItemCount` used for?

Reply:
There are some tests in the afterEach block on the describe block. This I have already discovered to be poor form from the "sibling tests share state" thread so will be a pattern of the past, but for now, this is the way the tests in this file are described.